common/beerocks/bwl/include/bwl/key_value_parser.h

```diff
@@ -0,0 +1,66 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2016-2019 Intel Corporation
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BWL_KEY_VALUE_PARSER_H_
+#define _BWL_KEY_VALUE_PARSER_H_
+
+#include <list>
+#include <string>
+#include <unordered_map>
+
+namespace bwl {
+
+/**
+ * @brief On KeyValueParser there are two types of object that can be parsed:
+ * parsed_line_t - Shall be used on key-value data with unique keys for each element, seperated by
+ * a space or a new line (Default seperator, can be modified on use). For example:
+ * "key1=value1 key2=value2 key3=value3 ..."
+ * --Or--
+ * "key1=value1
+ *  key2=value2
+ *  key3=value3
+ *  ..."
+ * 
+ * parsed_multiline_t - Shall be used in case there are 2 or more "parsed_line_t" which each "line"
+ * could have the same keys as the previous one. For example: 
+ * "key1=Line1value1 key2=Line1value2 key3=Line1value3 ...
+ *  key1=Line2value1 key2=Line2value2 key3=Line2value3 ...
+ *  key1=Line3value1 key2=Line3value2 key3=Line3value3 ...
+ *  ...""
+ */
+
+typedef std::unordered_map<std::string, std::string> parsed_line_t;
+typedef std::list<parsed_line_t> parsed_multiline_t;
+
+class KeyValueParser {
+protected:
+    static void parse_line(std::istringstream &iss, std::list<char> delimiter_list,
+                           parsed_line_t &parsed_line);
+
+    static void parse_multiline(std::istringstream &iss, std::list<char> delimiter_list,
+                                parsed_multiline_t &parsed_multiline);
+
+    static bool read_param_int(const std::string &key, parsed_line_t &obj, int64_t &value,
+                               bool ignore_unknown = false);
```

Comment:
As observed by @vitalybu in https://github.com/prplfoundation/prplMesh/pull/1491/commits/d8f21c02da9d711eb21ece722aa4889b5ff12e3c#r448829613 there's no need to specify the _int in the function name.
```suggestion
    static bool read_param(const std::string &key, parsed_line_t &obj, int64_t &value,
                               bool ignore_unknown = false);
```

Reply:
done.